src/config.ts

```diff
@@ -34,6 +34,7 @@ export const config = rt
     command: command,
     stackName: rt.String,
     workDir: rt.String,
+    projectName: rt.String,
```

Comment:
I think we can eliminate the `projectName` input entirely by relying on this line below:

```typescript
const projectSettings = await stack.workspace.projectSettings();
```

We should always be able to get the project name from the config.

Reply:
Ah, did you push that change? Right now, in the diff, I see the value in `Config` is always set to the empty string.
My goal is to eliminate partially initialized states in the `Config` object. As much as possible, that `Config` object should be immutable:
1. If at the time of construction, we can access the `projectName`, then that works.
2. But if not, I'd prefer passing the `projectName` field into the handlePullRequest` function.

From the diff I'm seeing, the config object is initialized with an empty string [here](https://github.com/pulumi/actions/pull/755/files#diff-c3095d5010e65c52737a98a5d618ea24049ebe90c8470752426081d70ed6e012R63) and that value is overwritten [here](https://github.com/pulumi/actions/pull/755/files#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR44-R46). I'd prefer if we could avoid this mutation.

Does that help? 😅 Sorry if that's unclear or inconsistent.

Reply:
Got it! I think it's easier to pass projectName as a field on the function as well (and also avoids having a variable with a different value, depending on time of calling). I'll look into it in a bit, commit the changes, and test it as well via the link you sent. I'll notify in here when I've tested it "for real" 👍